src/LodeRunner.Test/UnitTests/LoadTestConfigExtensionsTest.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using System.CommandLine;
+using System.CommandLine.Parsing;
+using LodeRunner.Core.CommandLine;
+using LodeRunner.Core.Extensions;
+using LodeRunner.Core.Models;
+using Xunit;
+
+namespace LodeRunner.Test.UnitTests
+{
+    /// <summary>
+    /// LoadTestConfigExtensions Unit Tests.
+    /// </summary>
+    public class LoadTestConfigExtensionsTest
+    {
+        /// <summary>
+        /// Creates MemberData parameter values for GetArgs Test Cases.
+        /// </summary>
+        /// <returns>Parameter values.</returns>
+        public static IEnumerable<object[]> GetLoadTestConfigParams()
+        {
+            yield return new object[] { new LoadTestConfig() { Server = new List<string> { "https://somerandomdomain.com" }, Files = new List<string> { "memory-baseline.json" }, Sleep = 1000 }, 6, "Validation for run once" };
+            yield return new object[] { new LoadTestConfig() { Server = new List<string> { "https://somerandomdomain.com" }, Files = new List<string> { "memory-baseline.json" }, Duration = 1, RunLoop = true }, 8, "Validation for run loop" };
+            yield return new object[] { new LoadTestConfig() { Server = new List<string> { "https://somerandomdomain.com", "https://someotherdomain.com" }, Files = new List<string> { "memory-baseline.json", "memory-benchmark.json" }, Duration = 1, RunLoop = true }, 10, "Validation for multiple servers and files listed" };
+        }
+
+        /// <summary>
+        /// Test success cases of GetArgs Extension to ensure conversion works as expected.
+        /// </summary>
+        /// <param name="loadTestConfig">The LoadTestConfig to convert to command line arguments.</param>
+        /// <param name="expectedArgsLength">The expected length of the generated args array.</param>
+        /// <param name="messageifFailed">The message to display if test failed.</param>
+        [Theory]
+        [Trait("Category", "Unit")]
+        [MemberData(nameof(GetLoadTestConfigParams))]
+        public void GetArgs_Success(LoadTestConfig loadTestConfig, int expectedArgsLength, string messageifFailed)
```

Comment:
Do we need a test method to check for failure as well?

Reply:
I debated doing this,but felt that the failure cases to check would be around testing the logic of how the command line parses arguments which we have many test cases on in the CommandLineArgs unit tests. However, I am realizing we should test GetArgs when a payloadPropertiesChanged list is not null (used in validation by the API).